### PR TITLE
CORTX-33445-TA-dtm0int1

### PR DESCRIPTION
**Pull request by kulrk**
Signed-off-by: kulrk <rohan.1.kulkarni@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 1]
```diff
@@ -200,10 +200,106 @@ def test_write_during_multi_m0d_restart(self):
...
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=log_file_prefix,
+                                      queue=que)
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info = resp[1]
+
+        time.sleep(self.delay)
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:235]: maybe just add a logger that u r waiting for xx secs for some ops

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 2]
```diff
@@ -200,10 +200,106 @@ def test_write_during_multi_m0d_restart(self):
...
+            check_proc_state=True,
+            k_value=self.kvalue)
+
+        self.log.info("Step 5: Wait for read Operation to complete.")
+        if proc_read_op.is_alive():
+            proc_read_op.join()
+        resp = que.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        # Assert if process restart failed
+        assert_utils.assert_true(resp_proc, "Failure observed during process restart/recovery")
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:257]: So we are not expecting any failures while m0d are getting restarted?

File: tests/dtm/test_multi_m0d_process_restart.py [hunk 3]
```diff
@@ -200,10 +200,106 @@ def test_write_during_multi_m0d_restart(self):
...
+        self.test_completed = True
+        self.log.info("ENDED: Verify READ during m0d restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-4XXXX")
```
Comment (sonalk0209) [inline tests/dtm/test_multi_m0d_process_restart.py:263]: make sure to add these before merging

File: tests/dtm/test_multi_m0d_process_restart.py (hunk 1 above)
Comment (kulrk) [reply tests/dtm/test_multi_m0d_process_restart.py:235]: added

File: tests/dtm/test_multi_m0d_process_restart.py (hunk 2 above)
Comment (kulrk) [reply tests/dtm/test_multi_m0d_process_restart.py:257]: we are expecting failures while restarting m0d, based on proc_restart, we are asserting in test

File: tests/dtm/test_multi_m0d_process_restart.py (hunk 3 above)
Comment (kulrk) [reply tests/dtm/test_multi_m0d_process_restart.py]: added.

File: tests/dtm/test_multi_m0d_process_restart.py (hunk 2 above)
Comment (sonalk0209) [reply tests/dtm/test_multi_m0d_process_restart.py:257]: Maybe not able to get it, @priyankaborawake can you check here?

**Review by sonalk0209 — APPROVED** [review]

File: tests/dtm/test_multi_m0d_process_restart.py (hunk 2 above)
Comment (priyankaborawake) [reply tests/dtm/test_multi_m0d_process_restart.py:257]: No We are not expecting any failure during m0d restart. 
as long as no of failures are within limits.

**Review by priyankaborawake — APPROVED** [review]

**Comment by kulrk** [comment]
image


collection log

**Review by apurwamohite — APPROVED** [review]

**Pull request merged**